src/components/NewSpendingForm/NewSpendingForm.jsx

```diff
@@ -1,61 +1,65 @@
 import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import * as Styled from './styles';
 import Button from '../Button';
+import { postSpendingPending } from '../../redux/actions';
 
 const NewSpendingForm = ({ onClose }) => {
   const categories = [
-    'Shopping',
-    'Entertainment',
-    'Car',
-    'Bills',
-    'Food',
-    'Home',
-    'Education',
-    'Other',
+    'shopping',
+    'entertainment',
+    'car',
+    'bills',
+    'food',
+    'home',
+    'education',
+    'other',
+  ];
+  const currencyData = [
+    { id: '1', value: 'byn', label: 'BYN' },
+    { id: '2', value: 'rub', label: 'RUB' },
+    { id: '3', value: 'usd', label: 'USD' },
+    { id: '4', value: 'eur', label: 'EUR' },
   ];
   const [category, setCategory] = useState(''); //eslint-disable-line
+  const [createdAt, setCreatedAt] = useState('');
+  const [amount, setAmount] = useState(0);
+  const [currency, setCurrency] = useState('');
+  const [note, setNote] = useState('');
+  const [labels, setLabels] = useState([]);
+  const dispatch = useDispatch();
+  const fieldNames = { category, createdAt, amount, currency, note, labels };
+
   const handleChangeCategory = (val) => {
     setCategory(val);
   };
 
-  const [periodStart, setPeriodStart] = useState('');
   const handleChangePeriodStart = (date) => {
-    setPeriodStart(date);
+    setCreatedAt(date);
   };
 
-  const [amount, setAmount] = useState(0);
   const handleChangeAmount = (val) => {
     setAmount(val);
   };
 
-  const currencyData = [
-    { id: '1', value: 'byn', label: 'BYN' },
-    { id: '2', value: 'rub', label: 'RUB' },
-    { id: '3', value: 'usd', label: 'USD' },
-    { id: '4', value: 'eur', label: 'EUR' },
-  ];
-  const [currency, setCurrency] = useState('');
   const handleChangeCurrency = (val) => {
     setCurrency(val);
   };
 
-  const [note, setNote] = useState('');
   const handleChangeNote = (val) => {
     setNote(val);
   };
 
-  const [labels, setLabels] = useState();
   const handleDeleteLabel = (deletingLabel) => {
     const newLabels = labels.filter((label) => label !== deletingLabel);
     setLabels(newLabels);
   };
+
   const handleChangeLabel = (val) => {
     setLabels([...labels, val]);
   };
 
-  const fieldNames = { category, periodStart, amount, currency, note, labels };
-
   const setSpendingValues = () => {
     return Object.fromEntries(Object.entries(fieldNames).filter(([key, value]) => value)); //eslint-disable-line
```

Comment:
Honestly, I don't get this function. Why don't you want to create simple object?